oars_gb_pkg/helpers/path_planning/short_course_planner.py

```diff
@@ -0,0 +1,145 @@
+'''
+This program implements the short course routing algorithm described in Section 5.1 of Roland Stelzer's Autonomous
+Sailboat Navigation paper. This algorithm uses a polar efficiency function to determine the efficiency of all possible
+angles that a boat can turn, and then compares the efficiencies to determine the optimal boat heading.
+
+Read the paper here: https://www.dora.dmu.ac.uk/bitstream/handle/2086/7364/thesis-optimized-300dpi.pdf
+'''
+import numpy as np
+from turtle import *
+from time import sleep
+
+class ShortCoursePlanner:
+
+    def polar_effeciency(self, windangle, alpha):
+        '''
+        Calculates the speed for a certain angle
+        :param windangle: the wind direction
+        :param alpha: angle being tested (relative to the wind angle)
+        :return: the vector components for the velocity for the angle
+        '''
+        # Convert angle to radians
+        angle = (windangle + alpha) * 180
+        alpha *= np.pi / 180
+
+        # Calculate magnitude using the polar effeciency function
+        mag = (1 - np.cos(alpha)) * (1 + 0.3 * np.cos(alpha)) / (1 - 0.5 * np.cos(alpha))
+
+        # Return vector components
+        return [mag * np.cos(angle), mag * np.sin(angle)]
+
+    # finds the maximum velocity and angle that can be achieved for each value of alpha for the right side and the left side
+    def optimum(self, windangle, path):
```

Comment:
Again, `wind_angle` :)

Also, this is probably another protected method. And since you're getting some data, `_get_optimum` or `_calc_optimum` might be more descriptive.